src/renderer/js/redux/reducers/app.js

```diff
@@ -1,22 +1,45 @@
-import * as types from "constants/action_types";
-import * as modalTypes from "constants/modal_types";
+// @flow
 
+import * as types from "../../constants/action_types";
+import * as modalTypes from "../../constants/modal_types";
+
+// $FlowIssue: Flow cannot find election
 const { remote } = require("electron");
+
 const application = remote.app;
 const win = remote.BrowserWindow.getFocusedWindow();
 
 const reducers = {};
-const defaultState = {
+type appState = {
+  isLoaded: boolean,
+  modal: mixed,
+  modalProps: mixed,
+  platform: string,
+  upgradeSkipped: boolean,
+  daemonVersionMatched: ?boolean,
+  daemonReady: boolean,
+  hasSignature: boolean,
+  badgeNumber: number,
+  volume: number,
+  downloadProgress?: number,
+  upgradeDownloading?: boolean,
+  upgradeDownloadComplete?: boolean,
+  checkUpgradeTimer?: mixed,
+  isUpgradeAvailable?: boolean,
+  isUpgradeSkipped?: boolean,
+  snackBar?: mixed,
```

Comment:
I don't remember what's put in this but I'm just hating on all of your usage of `mixed` at this point.

Reply:
me too, i'll figure it out